src/objects.h

```diff
@@ -293,17 +293,51 @@ GAP_STATIC_ASSERT(LAST_REAL_TNUM <= 254, "LAST_REAL_TNUM is too large");
 GAP_STATIC_ASSERT(LAST_COPYING_TNUM <= 254, "LAST_COPYING_TNUM is too large");
 #endif
 
+
 /****************************************************************************
 **
-** Object flags for use with SET_OBJ_FLAG() etc.
+*F  TEST_OBJ_FLAG(<obj>, <flag>) . . . . . . . . . . . . . . test object flag
+*F  SET_OBJ_FLAG(<obj>, <flag>) . . . . . . . . . . . . . . . set object flag
+*F  CLEAR_OBJ_FLAG(<obj>, <flag>) . . . . . . . . . . . . . clear object flag
+**
+**  These three macros test, set, and clear object flags, respectively.
+**  For non-immediate objects, these are simply the bag flags, see
+**  TEST_BAG_FLAG, SET_BAG_FLAG, CLEAR_BAG_FLAG.
 **
+**  For immediate objects, objects flags are always 0.
 */
+static inline uint8_t TEST_OBJ_FLAG(Obj obj, uint8_t flag)
+{
+    if (IS_BAG_REF(obj))
+        return TEST_BAG_FLAG(obj, flag);
+    else
+        return 0;
+}
+
+static inline void SET_OBJ_FLAG(Obj obj, uint8_t flag)
```

Comment:
Huh? They are used in `plist.c` and in `hpc/threadapi.c`.

And the handling of immediate objects is intentional (see also line 307 above), and the primary reason for this PR: so that we `*_BAG_FLAG` as a low-level gasman API, dealing with bags; and `*_OBJ_FLAG` as a higher-level API, dealing with arbitrary objects, including immediate objects.

Reply:
Not sure how my diff missed them. Weird.